### PR TITLE
Ensure correct color dialog is shown if light is off

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

If you have a light that only supports `color_temp` but is off, it would show the color picker rather than the temperature slider.

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/dialogs/more-info/components/lights/ha-more-info-view-light-color-picker.ts [hunk 1]
```diff
@@ -264,10 +264,18 @@ class MoreInfoViewLightColorPicker extends LitElement {
...
-        this.stateObj!.attributes.color_mode === LightColorMode.COLOR_TEMP
+      // If we have an active color mode, use it to determine what to render.
+      // Otherwise (e. g. light is off) fallback to the supported modes instead.
+      this._mode = this.stateObj!.attributes.color_mode
+        ? this.stateObj!.attributes.color_mode === LightColorMode.COLOR_TEMP
           ? LightColorMode.COLOR_TEMP
-          : "color";
+          : "color"
+        : this.stateObj!.attributes.supported_color_modes &&
+          this.stateObj!.attributes.supported_color_modes.indexOf(
```
Comment (bramkragten) [inline src/dialogs/more-info/components/lights/ha-more-info-view-light-color-picker.ts:274]: use `lightSupportsColorMode`

File: src/dialogs/more-info/components/lights/ha-more-info-view-light-color-picker.ts [hunk 2]
```diff
@@ -264,10 +264,18 @@ class MoreInfoViewLightColorPicker extends LitElement {
...
+            LightColorMode.COLOR_TEMP
+          ) >= 0
+        ? LightColorMode.COLOR_TEMP
+        : "color";
```
Comment (bramkragten) [inline src/dialogs/more-info/components/lights/ha-more-info-view-light-color-picker.ts:278]: Why default to color temp? Color is the first tab, so makes more sense to me to default to that one?

**Review by piitaya — COMMENTED** [review]
Why not just fallback to `this.modes[0]`? So the first tab will be selected.

**Comment by spacegaier** [comment]
Closing in favor of https://github.com/home-assistant/frontend/pull/15639

**Pull request closed without merging**